## Task_3

## Дано: 

Совместно с пиццерией «Да-да пицца» через мобильное приложение была
запущена маркетинговая кампания с подарком на группу гостей, которые ранее
сделали хотя бы один заказ в любой из пиццерий этой сети, но за последний месяц
не сделали ни одного заказа – 1000 человек. Компания предлагает при заказе от
200 рублей добавить в заказ бесплатный снек (например, картофель
по-деревенски, стоимость которого в меню - 100 рублей). В результате, ресторан
посетило 200 человек из тех, кому отправлено предложение, 100 из них
воспользовались предложением и взяли свой бесплатный картофель, 20 из них
сделали заказ на 400 и больше рублей.

Вопросы:

● Как можно оценить эффективность кампании?

● Успешна или убыточна акция?

● Какие показатели можно использовать для оценки этой кампании?

### Как можно оценить эффективность кампании?

(200 клиентов из 1000) 20% клиентов, которых мы возможно потеряли, использовали мобильное приложение и посетили пиццерию.

Исходя из отчёта Balance Grille при проведении компании по возврату ушедших клиентов 53% пользователей открывают таргетированное сообщение, 35% возвращаются и делают заказ, что даёт компании рост месячной выручки 3.7%.

Сколько пользователей открыли предложение неизвестно, но известно, что 20%  вернулись и сделали заказ, наши показатели в целом хуже чем у Balance Grille, но наблюдается положительная динамика.

Главные плюсы:

1) Churn rate повысился

2) Увеличилось общее впечатление об персонализированного предложения

3) Часть из них превратится в постоянных клиентов

4) Кто-то из них порекомендует наше приложение другим клиентам. Исходя из отчёта Balance Grille на одну рекомендацию в соц.сетях приходится 9 установок приложения, 5 новых клиентов, увеличение месячной выручки на 2.4%

Главные минусы:

1) Сложно сделать вывод о том, насколько предложение бесплатного картофеля является лучшим с точки зрения Customer expirience.

В целом, исходя из отчёта Balance Grille, подобная политика компании приводит к уменьшению числа людей, которые посещают ресторан 1 раз в месяц и к увеличению тех, кто посещает ресторан 3+ раза в месяц
Также происходит более плотная связь между клиентом и рестораном
Исходя из отчёта Sumo Sushi & Bento 80% людей делают решение о покупке, исходя из эмоций, что опять же даёт нам плюс, к той политике, которая проводит компания.

Также из отчёта известно, что на 71% процент увеличивается количество заказов, сравнивая неделю до компании по таргетированной рекламе и неделю во время компании), 6.6% процентов пользователей, которым было отправлено таргетированное сообщение, сделали заказ, месячная выручка увеличилась на 0.5%.

Это говорит нам о том, что компания по возврату клиентов "Да-да пицца" обладает конкурирующими с другими ресторанами показателями.

### Вывод:

В целом эффективность компании можно назвать успешной:

1) Увеличение всех ключевых показателей, имеющих отношение к Customer service и влияющих на выручку компании

2) Конкурентные значения показателей по сравнению с ресторанным рынком 

### Успешна или убыточна акция?

In [1]:
# Проанализируем цены на рынке наиболее известных фастфуд ресторанов в Санкт-Петербурге
# Ценники на картошку и минимальная стоимость съедобных блюд 
# (возьмём в расчёт, что покупатели не приобретали отдельно соусы, стикеры, салфетки и т.п.)

dodo_pizza_potato = 199
dodo_pizza_min_bon_aqua = 75

kfc_potato = 59
kfc_min_icecream = 39

bk_potato = 130
bk_min_cheese_bites = 69.99

black_star_potato = 150
black_star_min_bon_aqua = 99

dominos_potato = 189
dominos_min_drinks = 99

In [2]:
# Найдём отношение минимальной цены на блюдо в ресторанах к цене картошки

min_ratio = (dodo_pizza_min_bon_aqua/dodo_pizza_potato + \
             kfc_min_icecream/kfc_potato + \
             bk_min_cheese_bites/bk_potato + \
             black_star_min_bon_aqua/black_star_potato + \
             dominos_min_drinks/dominos_potato)/5
print(min_ratio)

0.5520191020914469


In [3]:
# Определим преполагаемую цену минимальной по цене позиции в "Да-да пицце", 
# исходя из полученного ранее отношения

da_da_pizza_potato = 100
da_da_pizza_estimated_min = round(da_da_pizza_potato*min_ratio)
print(da_da_pizza_estimated_min)

55


In [4]:
# В условии сказано, что ресторан посетило 200 человек.
# Предположим, что посетили = что-то купили
# 20 человек потратили 400+
# 100 человек воспользовались предложением и взяли свой бесплатный картофель => 80 человек потратили от 200 до 400 рублей
# Примем, что если человек не взял бесплатный картофель, значит потратил меньше 200 рублей = > 100 человек потратили до 200 рублей

customers_under_200 = 100
customers_between_200_400 = 80
customers_plus_400 = 20

In [5]:
# Сумма потраченных денег клиентов, потративших до 200 рублей

mean_check_under_200 = (200 + da_da_pizza_estimated_min)/2
sum_under_200 = mean_check_under_200*customers_under_200

In [6]:
# Сумма потраченных денег клиентов, потративших до от 200 до 400 рублей

mean_check_between_200_400 = (200 + 400)/2
sum_between_200_400 = mean_check_between_200_400*customers_between_200_400

In [7]:
# Сумма потраченных денег клиентов, потративших до от более 400 рублей
# Предположим негативный сценарий 
# Клиенты потратившие больше 400 рублей не тратили больше 500 рублей

mean_check_plus_400 = (400 + 500)/2
sum_plus_400 = mean_check_plus_400*customers_plus_400

In [8]:
# Общая сумма покупок и стоимость подаренного картофеля

sum_purchases = sum_under_200 + sum_between_200_400 + sum_plus_400
sum_potato = da_da_pizza_potato*100

In [9]:
# Рассчитаем при каком фудкосте(включает затраты на сырье, постоянные и дополнительные издрежки) блюд мы выйдем в ноль
# Не будем учитывать предполагаемую прибыль картофеля, которую компания могла получить при его реализации
# Примем, что фудкост для всех блюд одинаков (фудкост = (себестоимость продуктов / сумму выручки)*100)
# x - фудкост
# 1-x - часть стоимости блюда, представляющая чистую прибыль

# (1-x)*sum_purchases - x*sum_potato = 0
# sum_purchases - x*sum_purchases - x*sum_potato = 0
# sum_purchases = x*(sum_purchases + sum_potato)

x = sum_purchases / (sum_purchases + sum_potato)

print(f'''Итог: Фудкост, при котором бизнес не получит прибыли, но и не потеряет её равен {round(x*100,2)}%.''')

Итог: Фудкост, при котором бизнес не получит прибыли, но и не потеряет её равен 82.06%.


In [10]:
# Опираясь на данные из открытых источников рассмотрим оптимальную цену фудкоста: 
# https://rkeeper.ru/blog/zachem-schitat-natsenku-i-fudkost/   20-30%
# https://blog.quickresto.ru/chto-takoe-fudkost                25–40%
# https://open-s.info/company/news/kak-schitat-fudkost/        20-30%

# Предположим, что компания "Да-да пицца" не первый день на рынке, налажены цепочки поставок, 
# может позволить себе мобильное приложение
# Примем фудкост в районе 50%, условно включив туда ещё и затраты на маркетинговую компанию
# Рассчитаем успешность акции в денежном выражении

foodcost = 0.5

revenue_per_company = (1-foodcost)*sum_purchases - foodcost*sum_potato

print(f'''Итог: При фудкосте, равном {round(foodcost*100,2)}%, прибыль составит {revenue_per_company} руб.''')

Итог: При фудкосте, равном 50.0%, прибыль составит 17875.0 руб.


### Вывод:
При проведении маркетинговой компании, компания получила прибыль.

Также стоит учесть, что при проведении расчётов мы не учли низкую себестоимость картофеля, а также постэффект от данного мероприятия, который с большей долей вероятности будет иметь положительное влияние на бизнес (возврат ушедших клиентов, увеличение общего впечатления от компании "Да-да пицца", настраивании более глубокой связи между клиентом и безнесом, часть возвратившихся клиентов превратятся в постоянных).

### Какие показатели можно использовать для оценки этой кампании?

In [11]:
avg_check = sum_purchases/200
print(avg_check)

228.75


Так или иначе маркетинговая компания будет влиять на все типы ключевых показателей.

Наиболее подходящими для оценки эффективности компании следует принять:

Процент пользователей, ознакомившихся с предложением (Open_rate)

Процент пользователей, вернувшихся к услугам компании после ознакомления с предложением (Retention rate 20%)

Процент пользователей, воспользовавшийся предложением, из вернувшихся (50%)

Процент потерянных клиентов (Churn rate 80%)

Средний чек (avg check 228.75 руб.)

Отзывы клиентов, получивших предложение (NPS)

Выручка в ходе компании (Revenue)

Процент клиентов, перешедших в разряд постоянных (Conversion rate from light to heavy user)